### PR TITLE
Remove disable checkbox

### PR DESCRIPTION
**Pull request by janfaracik**
As part of https://github.com/jenkinsci/jenkins/pull/6485 the new job config screen will have a toggle for a project's state in the top right by default, so there's no need to duplicate this.

---

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Comment by janfaracik** [comment]
Hoping I've updated this correctly - does this look okay to you @timja?

File: pom.xml [hunk 1]
```diff
@@ -84,8 +84,8 @@
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.289.x</artifactId>
-                <version>1289.v5c4b_1c43511b_</version>
+                <artifactId>bom-weekly</artifactId>
```
Comment (timja) [inline pom.xml:87]:
```suggestion
                <artifactId>bom-2.346.1</artifactId>
```

**Comment by timja** [comment]
Required https://github.com/jenkinsci/matrix-project-plugin/pull/118

**Comment by jglick** [comment]
AFAICT this has no visible effect. Is it just code cleanup?

**Comment by jglick** [comment]
> There’s extra toggles when matrix projects are run to disable them without this

Where? I ran this PR with and without the `src/main/resources/hudson/matrix/MatrixProject/configure-entries.jelly` patch side by side and could not see any difference.

**Comment by janfaracik** [comment]
> > There’s extra toggles when matrix projects are run to disable them without this
> 
> Where? I ran this PR with and without the `src/main/resources/hudson/matrix/MatrixProject/configure-entries.jelly` patch side by side and could not see any difference.

Ah there's a workaround in Jenkins to hide the extra disable checkboxes supplied by plugins as it causes issues. By removing the checkbox in this plugin we can remove the workaround in Jenkins.

**Pull request merged**